crates/configuration/src/configuration.sql

```diff
@@ -1,242 +1,180 @@
-              'return_type',
-              r.scalar_type
-            )
-          ) as routines
-        from
-          (
-            select r.*,
-              -- These are the types we support, mapped to "standard" aliases.
-              -- We have a similar case expression above, the two needs to be in sync.
-              case r.data_type
-                when 'boolean' then 'boolean'
-                when 'smallint' then 'smallint'
-                when 'integer' then 'integer'
-                when 'bigint' then 'bigint'
-                when 'numeric' then 'numeric'
-                when 'real' then 'real'
-                when 'double precision' then 'double precision'
-                when 'text' then 'text'
-                when 'string' then 'string'
-                when 'character' then 'character'
-                when 'json' then 'json'
-                when 'jsonb' then 'jsonb'
-                when 'date' then 'date'
-                when 'time with time zone' then 'time with time zone'
-                when 'time without time zone' then 'time without time zone'
-                when 'timestamp with time zone' then 'timestamp with time zone'
-                when 'timestamp without time zone' then 'timestamp without time zone'
-                when 'uuid' then 'uuid'
-                else 'any'
-              end scalar_type
-            from hasura-development.chinook_sample.INFORMATION_SCHEMA.ROUTINES as r
-          ) as r
-        -- get the parameters count for a routine
-        left outer join UNNEST (
-          (select count(*) as count
-            from hasura-development.chinook_sample.INFORMATION_SCHEMA.PARAMETERS parameters
-            where r.specific_name = parameters.specific_name
-          )) parameters on ('true')
-        where routine_schema in ('pg_catalog', 'public')
-        -- include routines with only one parameter
-        and parameters.count = 1
-        and routine_type is null -- aggregate functions don't have a routine type
-        and r.scalar_type <> 'any'
-        group by r.scalar_type
-      ) as routines_by_type
-  ) as _aggregate_functions
+      END AS type,
+      CASE WHEN c.is_nullable = 'YES' THEN 'nullable' ELSE 'nonNullable' END AS nullable
+    )) AS column_info
+  FROM HASURA_DATABASE_NAME_PLACEHOLDER.INFORMATION_SCHEMA.TABLES AS t
+  JOIN HASURA_DATABASE_NAME_PLACEHOLDER.INFORMATION_SCHEMA.COLUMNS AS c
+    ON c.table_catalog = t.table_catalog
+    AND c.table_schema = t.table_schema
+    AND c.table_name = t.table_name
+  WHERE t.table_schema = 'HASURA_DATABASE_SCHEMA_PLACEHOLDER'
+),
+columns_struct AS (
+  SELECT
+    table_name,
+    table_catalog,
+    table_schema,
+    STRUCT(
+      STRING_AGG(
+        CONCAT('"', column_name, '":', column_info),
+        ','
+      ) AS columns_json
+    ) AS columns
+  FROM column_data
+  GROUP BY table_name, table_catalog, table_schema
+),
+relationship_data AS (
```

Comment:
TIL, BigQuery got foreign keys last year. https://cloud.google.com/blog/products/data-analytics/join-optimizations-with-bigquery-primary-and-foreign-keys